### PR TITLE
New port: lynis

### PR DESCRIPTION
**Pull request by newtonne**
- lynis is a security and system auditing tool to harden systems

- It's a shell script, but can be (and is) delivered as a package on
many *nix systems

- The script reads extra configuration which is included in the tarball.
I've put this in ${prefix}/libexec/lynis as this seemed like the best
place for it. Let me know if it should go elsewhere.

###### Type(s)

- [x] enhancement

###### Tested on

macOS 10.13.5
Xcode 9.4.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by pmetzger** [comment]
If you're meant to run it as a user, why libexec and not bin or sbin?

**Comment by newtonne** [comment]
Yeah, good point. I just tend to think of `libexec` as more of a directory for "helper" files than `bin` or `sbin`. I'm actually now thinking that `etc` would be the best fit for them.

File: security/lynis/Portfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+PortSystem          1.0
+
+name                lynis
+version             2.6.6
+
+categories          security
+license             GPL-3
+maintainers         {gmail.com:newtonne.github @newtonne} openmaintainer
+platforms           darwin
+supported_archs     x86_64
```
Comment (mf2k) [inline security/lynis/Portfile]: Based on the comment below that this is a shell script, this line is not correct. It should be:

`supported_archs noarch`

File: security/lynis/Portfile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    Linux, macOS, or Unix-based operating system. \
+                    It performs an extensive health scan of your systems to \
+                    support system hardening and compliance testing. The \
+                    project is open source software with the GPL license and \
+                    available since 2007.
+
+homepage            https://cisofy.com/lynis/
+
+master_sites        https://downloads.cisofy.com/${name}
+distname            ${name}-${version}
```
Comment (mf2k) [inline security/lynis/Portfile]: This is the default value and so this line should be removed.

**Comment by pmetzger** [comment]
Perhaps the main script goes somewhere like sbin and things that it needs go into libsomething?

**Comment by newtonne** [comment]
@mf2k Thanks. Fixed both in latest commit.

@pmetzger I still think `etc` is the best place for these. As I mention in the commit message, they're not executables, they're not libraries, they're miscellaneous config files. However, if you really don't want them in there, I'm happy to move them. I also think `bin` is better suited than `sbin` for the lynis executable, since whilst it can be run as root for extra functionality, it can also be run as an unprivileged user.

**Comment by pmetzger** [comment]
Isn't the main shell script a shell script and thus executable?

If there's a set of configs that go in etc/, they should go together in a subdir named after the package (which is how this is generally done in MacPorts.)

**Comment by pmetzger** [comment]
BTW, sorry to seem grumpy. Thanks for your contribution to MacPorts, and your patience with getting your contributions merged.

**Comment by newtonne** [comment]
No problem. I appreciate the gratitude.

> Isn't the main shell script a shell script and thus executable?

Yeah, it is. I'm copying the main shell script, `lynis`, into the `${prefix}/bin` directory:
`xinstall -m 755 ${worksrcpath}/${name} ${destroot}${prefix}/bin/${name}`

> If there's a set of configs that go in etc/, they should go together in a subdir named after the package (which is how this is generally done in MacPorts.)

Indeed, in the destroot step, I am copying them to `${target_dir}` which is defined as:
`set target_dir ${prefix}/etc/${name}`

**Comment by pmetzger** [comment]
Gotcha. Then I think this is fine.

**Comment by pmetzger** [comment]
Thanks so much, @newtonne!

**Comment by ryandesign** [comment]
Next time, please squash before merging.

**Pull request merged**